src/stavenote.ts

```diff
@@ -14,22 +14,95 @@ import { Flow } from './tables';
 import { BoundingBox } from './boundingbox';
 import { Stem } from './stem';
 import { NoteHead } from './notehead';
-import { StemmableNote } from './stemmablenote';
+import { StemmableNote, StemStruct } from './stemmablenote';
 import { Modifier } from './modifier';
 import { Dot } from './dot';
+import { KeyProps, ModifierContextState } from './types/common';
+import { Beam } from './beam';
+import { ModifierContext } from './modifiercontext';
+import { Font } from './font';
+import { Fraction } from './fraction';
+import { ElementStyle } from './element';
+import { Stave } from './stave';
+export interface StaveNoteGetModifierStartXYOptions {
+  forceFlagRight: boolean;
+}
+
+export interface StaveNoteHeadBounds {
+  y_top: number;
+  y_bottom: number;
+  displaced_x: number;
+  non_displaced_x: number;
+  highest_line: number;
+  lowest_line: number;
+  highest_displaced_line: number;
+  lowest_displaced_line: number;
+  highest_non_displaced_line: number;
+  lowest_non_displaced_line: number;
+}
+
+export interface StaveNoteFormatSettings {
+  line: number;
+  maxLine: number;
+  minLine: number;
+  isrest: boolean;
+  stemDirection: number;
+  stemMax: number;
+  stemMin: number;
+  voice_shift: number;
+  is_displaced: boolean;
+  note: StaveNote;
+}
+
+export interface StaveNoteStruct {
```

Comment:
Wow, this is a lot. Are all of these really the options for `StaveNote`? I can't imagine why we have `font`, `subscript`, `superscript`, etc. here.

Reply:
I have left only the elements used and I have also extended it from `NoteStruct` to avoid duplication. If we need some element more in the migration we can always add it :)

Reply:
Thanks, looks way better -- can you eyeball the tests and see if everything renders correctly?